security_standards.md

```diff
@@ -0,0 +1,114 @@
+
+
+
+
+
+# Security Standard.
+
+# Background
+
+Having a secure approach to development has never been so important.
+
+The way we build software and systems is rapidly evolving, becoming more and more automated and integrated. This results in a need to have some standards and guidance around security. To save our time looking up and picking ways and more importantly to keep things uniform so things are done in the best way across the board
+
+This guidance will help you understand the security implications of modern code development and deployment practices.
+
+
+
+## Principles
+
+Secure development principles
+
+**1.** **Secure development is everyone's concern**
+
+Genuine security benefits can only be released when delivery teams weave security into their everyday working practices.
+
+**2.**  **Keep your security knowledge sharp**
+
+Creating code that is capable of withstanding attack requires an understanding of attack types and of defensive security practices.
+
+**3.**  **Produce clean & maintainable code**
+
+If your code lacks consistency, is poorly laid out and undocumented, you're adding to the overall complexity of your system.
+
+**4.**  **Secure your development environment**
+
+There is sometimes a perceived conflict between security and usability. This situation is highlighted in the case of end user devices and the environments used to support software development.
+
+**5.**  **Protect your code repository**
+
+Your code is only as secure as the systems used to create it. As the central point at which your code is stored and managed, it's crucial that the repository is sufficiently secure.
+
+**6.**  **Secure the build and deployment pipeline**
+
+Continuous integration, delivery and deployment are modern approaches to the building, testing and deployment of IT systems.
+
+**7.** **Continually test your security**
+
+Security testing can be manual, but it can also be automated.
+
+**8.** **Plan for security flaws**
+
+All but the very simplest software is likely to contain bugs, some of which may have a security impact.
+
+For more detailed information on the above principals see here:
+
+[https://www.ncsc.gov.uk/collection/developers-collection?curPage=/collection/developers-collection/principles](https://www.ncsc.gov.uk/collection/developers-collection?curPage=/collection/developers-collection/principles)
+
+
+
+## Standards
+
+Coding security standards are defined in this document:
+
+- [OWASP Secure Coding Practices](/owasp_secure_coding_practices.md)
+
+It is taken from the OWASP here:
+
+[https://www.owasp.org/index.php/OWASP_Secure_Coding_Practices_-_Quick_Reference_Guide](https://www.owasp.org/index.php/OWASP_Secure_Coding_Practices_-_Quick_Reference_Guide)
+
+
+
+## Guidance
```

Comment:
Similar suggestion to the principles, I'd recommend moving this to its own `security_guide.md` document.

That might help break things up a bit 😁